opentelemetry-exporter-geneva/geneva-uploader/src/client.rs

```diff
@@ -89,17 +99,49 @@ impl GenevaClient {
             .iter()
             .flat_map(|resource_log| resource_log.scope_logs.iter())
             .flat_map(|scope_log| scope_log.log_records.iter());
+        // TODO: Investigate using tokio::spawn_blocking for event encoding to avoid blocking
+        // the async executor thread for CPU-intensive work.
         let blobs = self.encoder.encode_log_batch(log_iter, &self.metadata);
-        for (event_name, encoded_blob, _row_count) in blobs {
-            // TODO - log encoded_blob for debugging
-            let compressed_blob = lz4_chunked_compression(&encoded_blob)
-                .map_err(|e| format!("LZ4 compression failed: {e}"))?;
-            // TODO - log compressed_blob for debugging
-            let event_version = "Ver2v0"; // TODO - find the actual value to be populated
-            self.uploader
-                .upload(compressed_blob, &event_name, event_version)
-                .await
-                .map_err(|e| format!("Geneva upload failed: {e}"))?;
+
+        // create an iterator that yields futures for each upload
+        let upload_futures = blobs
+            .into_iter()
+            .map(|(event_name, encoded_blob, _row_count)| {
+                let uploader = Arc::clone(&self.uploader);
+                let event_version = "Ver2v0"; // TODO - find the actual value to be populated
+                async move {
+                    // TODO: Investigate using tokio::spawn_blocking for LZ4 compression to avoid blocking
+                    // the async executor thread for CPU-intensive work.
+                    let compressed_blob = match lz4_chunked_compression(&encoded_blob) {
+                        Ok(blob) => blob,
+                        Err(e) => {
+                            return Err(format!("LZ4 compression failed: {e} Event: {event_name}"))
+                        }
+                    };
+                    match uploader
+                        .upload(compressed_blob, &event_name, event_version)
+                        .await
+                    {
+                        Ok(_) => Ok(()),
+                        Err(e) => Err(format!("Geneva upload failed: {e} Event: {event_name}")),
+                    }
+                }
+            });
+        // Execute uploads concurrently with configurable concurrency
+        let results: Vec<Result<(), String>> = stream::iter(upload_futures)
```

Comment:
We could avoid this intermediate allocation of `Vec<Result<(), String>>` by doing all the operations in one iterator chain:

```rust
let errors: Vec<String> = stream::iter(upload_futures)
    .buffer_unordered(self.max_concurrent_uploads)
    .filter_map(|result| async move { result.err() })
    .collect()
    .await;
```